addons/project_sale_expense/models/project_project.py

```diff
@@ -13,24 +13,28 @@ def _get_expenses_profitability_items(self, with_action=True):
         if not self.analytic_account_id:
             return {}
         can_see_expense = with_action and self.user_has_groups('hr_expense.group_hr_expense_team_approver')
-        expenses_read_group = self.env['hr.expense'].sudo()._read_group(
-            [
-                ('analytic_distribution_stored_char', '=ilike', f'%"{self.analytic_account_id.id}":%'),
-                ('is_refused', '=', False),
-                ('state', 'in', ['approved', 'done']),
-            ],
-            ['sale_order_id', 'product_id', 'ids:array_agg(id)', 'untaxed_amount'],
-            ['sale_order_id', 'product_id'],
-            lazy=False,
-        )
+        query = self.env['hr.expense']._search([('is_refused', '=', False), ('state', 'in', ['approved', 'done'])])
+        query.add_where('hr_expense.analytic_distribution ? %s', [str(self.analytic_account_id.id)])
+        from_clause, where_clause, where_param = query.get_sql()
+        query = """
+            SELECT sale_order_id, product_id, array_agg(id) as ids, SUM(untaxed_amount) as untaxed_amount
+            FROM %(from)s
+            WHERE %(where)s
+            GROUP BY sale_order_id, product_id
```

Comment:
Same as in `project_hr_expense`, but you can add the group by after by getting the query with `.select()` insted of rebuilding everything